posthog/tasks/alerts/test/test_checks.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+from typing import Any, Optional
+from unittest.mock import MagicMock, patch
+
+from freezegun import freeze_time
+
+from posthog.models.instance_setting import set_instance_setting
+from posthog.tasks.alerts.checks import check_all_alerts
+from posthog.test.base import APIBaseTest, _create_event, flush_persons_and_events, ClickhouseDestroyTablesMixin
+from posthog.api.test.dashboards import DashboardAPI
+from posthog.schema import ChartDisplayType, EventsNode, TrendsQuery, TrendsFilter
+from posthog.tasks.test.utils_email_tests import mock_email_messages
+
+
+@freeze_time("2024-06-02T08:55:00.000Z")
+@patch("posthog.tasks.alerts.checks.EmailMessage")
+class TestDetectAlertsAnomaliesTasks(APIBaseTest, ClickhouseDestroyTablesMixin):
+    def setUp(self) -> None:
+        super().setUp()
+        set_instance_setting("EMAIL_HOST", "fake_host")
+        set_instance_setting("EMAIL_ENABLED", True)
+        self.settings(CELERY_TASK_ALWAYS_EAGER=True)
```

Comment:
Ah, one other thing: Do not use `always_eager` in tests. The preferred way to test here is mocking. Mock the task and check that it is triggered. Test what the task does by testing what it calls (hence pulling out code out of the task into separate functions).

Reply:
I don't know, why not? I prefer tests without mocks over tests with mocks as long as the former is not very difficult to achieve. They cover more, less chances to miss a spot in the functionality.

Reply:
If you don't know and want to know why, please read the docs.

> The eager mode enabled by the [task_always_eager](https://celeryproject.readthedocs.io/zh-cn/latest/userguide/configuration.html#std:setting-task_always_eager) setting is by definition not suitable for unit tests.
>
> When testing with eager mode you are only testing an emulation of what happens in a worker, and there are many discrepancies between the emulation and what happens in reality.

Reply:
Could you please add rationale to your change requests in the future? I couldn't know if it was your opinion or a documentation guideline, right? I was assuming the former.

Reply:
I see, sorry about that! I felt stating it as "preferred way" was enough to not make it seems as just my opinion.

Reply:
Fixed